osu.Game/Database/RealmContextFactory.cs

```diff
@@ -96,6 +96,9 @@ public RealmContextFactory(Storage storage, string filename, IDatabaseContextFac
                 Filename += realm_extension;
 
             cleanupPendingDeletions();
+
+            // run after the above operation to ensure realm is already in a good (fully migrated) state.
```

Comment:
Confused by this comment, especially the mention of "fully migrated". Did this mean to say "fully cleaned up"?

Reply:
I meant the realm level migrations, which are separate from data migrations (see `onMigration` stuff).

Reply:
I've rewritten the comments around this code to hopefully be easier to follow.